external-plugins/test-subset/plugin/handler/handler.go

```diff
@@ -0,0 +1,371 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	k8s_v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "sigs.k8s.io/prow/pkg/client/clientset/versioned/typed/prowjobs/v1"
+	"sigs.k8s.io/prow/pkg/config"
+	gitv2 "sigs.k8s.io/prow/pkg/git/v2"
+	"sigs.k8s.io/prow/pkg/github"
+	"sigs.k8s.io/prow/pkg/pjutil"
+
+	pi_github "kubevirt.io/project-infra/robots/pkg/github"
+)
+
+var log *logrus.Logger
+var testSubsetCommentRe = regexp.MustCompile(`^/test-subset (\S+) (\S+|\(.*\))$`)
+
+func init() {
+	log = logrus.New()
+	log.SetOutput(os.Stdout)
+}
```

Comment:
Please use structured logging:

```suggestion
	log.SetFormatter(&logrus.JSONFormatter{})
}
```